### ISSUE TITLE
Implement general callback to perform an action after some epochs of no improvement on a monitored metric

### ISSUE THREAD

**Issue by cengh** [fchollet/keras#4857]
Currently, the callbacks `EarlyStopping` and `ReduceLROnPlateau` perform the same general function of carrying out some action after some number of epochs of no improvement on a monitored metric. These classes could be rewritten as subclasses of a more general class that monitors a quantity of interest and calls an `action()` function once this quantity ceases to improve. This would make it easy to implement other callbacks that are dependent on training progress in the future.
